internal/controller/model_nodeport.go

```diff
@@ -40,27 +52,61 @@ func NewNodePortLoadBalancerModelGenerator(
 	}
 }
 
-func (g *NodePortLoadBalancerModelGenerator) getDestinationAddresses() (result []string, err error) {
+func isValidIp(ipString string) (err error) {
+	ipParsed := net.ParseIP(ipString)
+	if ipParsed != nil {
+		return nil
+	}
+	return ErrInvalidIpAddress
+}
+
+func isIPv4Address(ipString string) (isIPv4 bool, err error) {
+	return strings.Count(ipString, ":") < 2, isValidIp(ipString)
+}
+
+func isIPv6Address(ipString string) (isIPv6 bool, err error) {
+	return strings.Count(ipString, ":") >= 2, isValidIp(ipString)
+}
+
+func (g *NodePortLoadBalancerModelGenerator) getDestinationAddresses() (addresses_v4 []string, addresses_v6 []string, err error) {
 	nodes, err := g.nodes.List(labels.Everything())
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	result = []string{}
 	for _, node := range nodes {
 		for _, addr := range node.Status.Addresses {
 			if addr.Type != corev1.NodeInternalIP {
 				continue
 			}
-			result = append(result, addr.Address)
+
+			nodeAddressIsIPv4, err := isIPv4Address(addr.Address)
+			if err != nil {
+				return nil, nil, err
+			}
+			if nodeAddressIsIPv4 {
+				addresses_v4 = append(addresses_v4, addr.Address)
+			}
+
+			nodeAddressIsIPv6, err := isIPv6Address(addr.Address)
+			if err != nil {
+				return nil, nil, err
+			}
+			if nodeAddressIsIPv6 {
+				addresses_v6 = append(addresses_v6, addr.Address)
+			}
```

Comment:
Do not exit the loop on a malformed IP address. Just skip that particular address.

Reply:
Thanks, changed.